### PR TITLE
add "dotproduct" distance metric

### PR DESCRIPTION
**Pull request by arnej27959**
@kkraune @jobergum  @geirst 
please review

### CODE REVIEW

**Review by jobergum — APPROVED** [review]

**Comment by jobergum** [comment]
Great :)

File: en/reference/schema-reference.html [hunk 1]
```diff
@@ -2768,6 +2887,61 @@ <h3 id="angular">angular</h3>
...
+  <em>prenormalized-angular</em> is used to build the HNSW index.
+  For details, see
+  <a href="https://towardsdatascience.com/maximum-inner-product-search-using-nearest-neighbor-search-algorithms-c125d24777ef">
+  this high level guide</a> based on
+  <a href="https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/XboxInnerProduct.pdf">
+  section 3.1 Order Preserving Transformations in this paper</a>.
+</p>
+<p>
+  Note that the <em>distance</em> and <em>closeness</em> rank-features will not have the
+  usual semantic meanings when using the <em>dotproduct</em> keyword.
```
Comment (geirst) [inline en/reference/schema-reference.html:2908]:
```suggestion
  usual semantic meanings when using the <em>dotproduct</em> distance metric.
```

File: en/reference/schema-reference.html [hunk 2]
```diff
@@ -2768,6 +2887,61 @@ <h3 id="angular">angular</h3>
...
+  In particular, "closeness" will just return the dot product
```
Comment (geirst) [inline en/reference/schema-reference.html:2909]:
```suggestion
  In particular, <em>closeness</em> will just return the dot product
```

File: en/reference/schema-reference.html [hunk 3]
```diff
@@ -2768,6 +2887,61 @@ <h3 id="angular">angular</h3>
...
+            <mo>(</mo>
+            <msub><mi>x</mi><mi>i</mi></msub>
+            <mo>*</mo>
+            <msub><mi>y</mi><mi>i</mi></msub>
+            <mo>)</mo>
+          </mrow>
+        </mrow>
+    </math>
+  which may have any negative
+  or positive value, and "distance" is just the negative dot product.
```
Comment (geirst) [inline en/reference/schema-reference.html:2923]:
```suggestion
  or positive value, and <em>distance</em> is just the negative dot product.
```

File: en/reference/schema-reference.html [hunk 4]
```diff
@@ -2768,6 +2887,61 @@ <h3 id="angular">angular</h3>
...
+  If a normalized closeness in range <code>[0,1]</code> is needed, an appropriate
+  <a href="https://en.wikipedia.org/wiki/Sigmoid_function">sigmoid function</a> must be applied.
+  For example, if your attribute is named "foobar", and the maximum dotproduct seen is
+  around 4000, the expression
+  <code>sigmoid(0.001*closeness(field,foobar))</code>
+  could be a possible choice.
+</p>
+<p>
+  The <em>dotproduct</em> keyword is very useful for some vectorization
```
Comment (geirst) [inline en/reference/schema-reference.html:2932]:
```suggestion
  The <em>dotproduct</em> distance metric is very useful for some vectorization
```

**Review by geirst — APPROVED** [review]
👍

File: en/reference/schema-reference.html [hunk 5]
```diff
@@ -2768,6 +2887,61 @@ <h3 id="angular">angular</h3>
   L2 norm and use the <code>prenormalized-angular</code> metric instead; but note that returned
   distance and closeness will be differerent.
 
+<h3 id="dotproduct">dotproduct</h3>
+<p>
+  The <em>dotproduct</em> distance metric is used to <em>mathematically
+  transform</em> a "maximum inner product" search into a form where
+  it can be solved by nearest neighbor search, where the dotproduct
+  is used as a score directly (large positive dotprodcuts are
```
Comment (vekterli) [inline en/reference/schema-reference.html:2895]:
```suggestion
  is used as a score directly (large positive dotproducts are
```

**Review by kkraune — COMMENTED** [review]
Great!! 💯

**Pull request merged**